bot/worker.py

```diff
@@ -0,0 +1,184 @@
+"""Main module of the Dota 2 subreddit Responses Bot.
+
+The main body of the script is running in this file. The comments are loaded from the subreddit
+and the script checks if the thing is a response from Dota 2. If it is, a proper thing is
+prepared. The thing is posted as a reply to the original post on Reddit.
+
+Proper logging is provided - saved to 2 files as standard output and errors.
+"""
+import string
+
+from praw.models import Comment
+
+import config
+from bot import account
+from util.caching import get_cache_api
+from util.database.database import db_api
+from util.logger import logger
+
+__author__ = 'Jonarzz'
+__maintainer__ = 'MePsyDuck'
+
+cache_api = get_cache_api()
+
+
+def work():
+    """Main method executing the script.
+
+    It connects to an account, loads dictionaries from proper files (declared in config file).
+    Afterwards it executes process_comments method with proper arguments passed.
+    """
+
+    reddit = account.get_account()
+    logger.info('Connected to Reddit account : ' + config.USERNAME)
+
+    comment_stream = reddit.subreddit(config.SUBREDDIT).stream.comments(pause_after=-1)
+    submission_stream = reddit.subreddit(config.SUBREDDIT).stream.submissions(pause_after=-1)
+    while True:
+        for comment in comment_stream:
+            if comment is None:
+                break
+            process_thing(reddit, comment)
+        for submission in submission_stream:
+            if submission is None:
+                break
+            process_thing(reddit, submission)
+
+
+def process_thing(reddit, thing):
+    """Method used to check all the comments in a submission and add replies if they are responses.
+
+    PRAW generates past ~100 comments on the first iteration. Then the loop only runs if there is a new thing added to
+    the comments stream. This also means that once PRAW is up and running, after the initial comments list it won't
+    generate any duplicate comments.
+
+    However, just as a safeguard, Caching is used to store thing ids as they are processed for the first time.
+    Otherwise, when the bot is restarted it might reply twice to same comments. If thing id is in the already present
+    in the cache_api, then it is ignored, else processed and added to the cache_api.
+    * Self comments are ignored.
+    * It is prepared for comparision to the responses in dictionary.
+    * If the thing is not on the excluded responses list (loaded from config) and if it is in the responses db or
+    specific responses list, a reply thing is prepared and posted.
+    """
```

Comment:
Lacks params info.

Reply:
Fixed in 41e1416fd2c7051547d1ef4807ac17741893a239